.github/workflows/nightly_test.yml

```diff
@@ -0,0 +1,205 @@
+name: Nightly Test Corso
+env:
+  IMAGE_NAME: ghcr.io/alcionai/corso
+on:
+  workflow_dispatch:
+
+  pull_request:
+
+  push:
+    branches: [main]
+    tags: ["v*.*.*"]
+
+  repository_dispatch:
+    types: [ok-to-test-command]
+
+permissions:
+  # required to retrieve AWS credentials
+  id-token: write
+  contents: write
+  packages: write
+  pull-requests: read
+
+# cancel currently running jobs if a new version of the branch is pushed
+concurrency:
+  group: ci-${{ github.workflow }}-${{ github.ref }}
```

Comment:
This should probably use a separate concurrency group than the one that the ci uses.